ledger/src/shred.rs

```diff
@@ -607,12 +609,25 @@ impl Shredder {
         fec_rate: f32,
         data_shred_batch: &[Shred],
         version: u16,
+        max_coding_shreds: bool,
     ) -> Vec<Shred> {
         assert!(!data_shred_batch.is_empty());
         if fec_rate != 0.0 {
             let num_data = data_shred_batch.len();
             // always generate at least 1 coding shred even if the fec_rate doesn't allow it
-            let num_coding = Self::calculate_num_coding_shreds(num_data as f32, fec_rate);
+            let shred_count = if max_coding_shreds {
+                MAX_DATA_SHREDS_PER_FEC_BLOCK as usize
+            } else {
+                num_data
+            };
+            let num_coding = Self::calculate_num_coding_shreds(shred_count as f32, fec_rate);
+            if num_coding > num_data {
+                trace!(
```

Comment:
So the only time this trace should trigger is if we have some remaining shreds in a big batch? So for instance if I have 33 data shreds, then the first 32 go into the first batch, and then the last data shred will be padded with 32 coding shreds?

Can we add a test outlining this expected behavior?

Reply:
@carllin , yes that's the case it's covering. Do you think it makes sense to generate max coding shreds even for batches that have < 32 data shreds. For example, say the TPS is such that every  batch has around 17 data shreds. Right now it'll generate that many coding shreds.

I'll add the test to cover the conditions..

Reply:
Added the test